node/main.go

```diff
@@ -66,26 +67,38 @@ func main() {
 
 	safePoint := &tikv.SafePoint{}
 	db := createDB("kv", safePoint)
-
-	regionOpts := tikv.RegionOptions{
-		StoreAddr:  *storeAddr,
-		PDAddr:     *pdAddr,
-		RegionSize: *regionSize,
-	}
-	rm := tikv.NewStandAloneRegionManager(db, regionOpts)
 	bundle := &mvcc.DBBundle{
 		DB:            db,
 		LockStore:     lockstore.NewMemStore(8 << 20),
 		RollbackStore: lockstore.NewMemStore(256 << 10),
 	}
-	var dbWriter mvcc.DBWriter
+
+	var innerServer server.InnerServer
 	if *raft {
-		dbWriter = raftstore.NewDBWriter(bundle, raftstore.NewDefaultConfig())
+		innerServer = newRaftServerBuilder(bundle, safePoint)
 	} else {
-		dbWriter = tikv.NewDBWriter(bundle, safePoint)
+		innerServer = newStandAlongBuilder(bundle, safePoint)
+	}
+
+	pdClient, err := pd.NewClient(*pdAddr, "")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	builderCtx := server.Context{
+		PDClient: pdClient,
+	}
+
+	if err := innerServer.Setup(&builderCtx); err != nil {
+		log.Fatal(err)
+	}
+
+	store := tikv.NewMVCCStore(bundle, *dbPath, safePoint, innerServer.GetDBWriter())
+	tikvServer := tikv.NewServer(innerServer.GetRegionManager(), store)
```

Comment:
I think make the inner server as a field of `tikvServer` is better, so we can implement `TiKVServer` methods like
`func (svr *Server) Raft(tikvpb.Tikv_RaftServer) error {` with inner server.

The interface `InnerServer` can be defined in `tikv` package, the implementation of stand alone can be defined in the `tikv` package too, and `RaftInnerServer` can be defined in `raftstore` package. So we don't need the `server` package.